ci/envoy-sync-receive.sh

```diff
@@ -0,0 +1,124 @@
+#
+# It merges the specified branch from the upstream envoyproxy/envoy repository
+# into the current branch in the current working directory.  It is assumed that
+# the calling workflow has already checked out the destination repository and
+# switched to the destination branch, in the current working directory before
+# invoking us.
+#
+# - If the merge is successful, it:
+#  - pushes the feature branch to the repository
+#  - creates the associated pull request if it doesn't already exist
+#  - closes the associated issue if it already exists
+#
+# -If the merge is unsuccessful, it:
+#  - leaves the associated pull request untouched if it already exists
+#  - creates the associated issue issue if it doesn't already exist
+#  - adds a comment on the associated issue to describe the merge fail
+#
+
+set -xeuo pipefail
+
+notice() { printf "::notice:: %s\n" "$1"; }
+
+SCRATCH="$(mktemp -d)"
+cleanup() {
+    local savexit=$?
+    rm -rf -- "${SCRATCH}"
+    exit "${savexit}"
+}
+trap 'cleanup' EXIT
+
+SRC_REPO_URL="https://github.com/envoyproxy/envoy.git"
+SRC_REPO_PATH="${SRC_REPO_URL/#*github.com?/}"
+SRC_REPO_PATH="${SRC_REPO_PATH/%.git/}"
+SRC_BRANCH_NAME="$1"
+SRC_HEAD_SHA="$(git ls-remote "${SRC_REPO_URL}" | awk "/\srefs\/heads\/${SRC_BRANCH_NAME/\//\\\/}$/{print \$1}")"
+
+DST_REPO_URL=$(git remote get-url origin)
+DST_REPO_PATH="${DST_REPO_URL/#*github.com?/}"
+DST_REPO_PATH="${DST_REPO_PATH/%.git/}"
+DST_BRANCH_NAME=$(git branch --show-current)
+DST_HEAD_SHA=$(git rev-parse HEAD)
+
+# Add the remote upstream repo and fetch the specified branch
+git remote remove upstream &> /dev/null || true
+git remote add -f -t "${SRC_BRANCH_NAME}" upstream "${SRC_REPO_URL}"
+
+# Compose text for pull request or issue title
+TITLE="auto-merge ${SRC_REPO_PATH}[${SRC_BRANCH_NAME}] "
+TITLE+="into ${DST_REPO_PATH}[${DST_BRANCH_NAME}]"
+
+# Create a new branch name for the merge. Deliberately don't include
+# any commit hash or timestamp in the name to ensure it is repeatable.
+# This ensures that each time we get invoked, due to an upstream change,
+# we accumulate the changes in the same branch and pull request, rather
+# than creating new ones that superceed the old one(s) each time.
+DST_NEW_BRANCH_NAME="auto-merge-$(echo "${SRC_BRANCH_NAME}" | tr /. -)"
+
+# Set the default remote for the gh command
+gh repo set-default "${DST_REPO_PATH}"
+
+# Perform the merge using --no-ff option to force creating a merge commit
+if git merge --no-ff --log=10000 --signoff -m "${TITLE}" \
+                "upstream/${SRC_BRANCH_NAME}" > "${SCRATCH}/mergeout"; then
+    DST_NEW_HEAD_SHA="$(git rev-parse HEAD)"
+    if [[ "${DST_NEW_HEAD_SHA}" != "${DST_HEAD_SHA}" ]]; then
+        git push --force origin "HEAD:${DST_NEW_BRANCH_NAME}"
+        PR_COUNT=$(gh pr list --head "${DST_NEW_BRANCH_NAME}" \
+                              --base "${DST_BRANCH_NAME}" \
+                              --state open | wc -l)
+        if [[ ${PR_COUNT} == 0 ]]; then
```

Comment:
nit: I'd surrender both parts with quotes, since we are not comparing numbers but strings, same below with ISSUE_COUNT